plugins/code-coverage/src/components/CoverageHistoryChart/CoverageHistoryChart.tsx

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import {
+  Box,
+  Card,
+  CardContent,
+  MenuItem,
+  Select,
+  Typography,
+} from '@material-ui/core';
+import {
+  LineChart,
+  XAxis,
+  YAxis,
+  Tooltip,
+  Legend,
+  Line,
+  CartesianGrid,
+  ResponsiveContainer,
+} from 'recharts';
+import { useAsync } from 'react-use';
+import { useApi } from '@backstage/core-api';
+import { useEntity } from '@backstage/plugin-catalog-react';
+import { codeCoverageApiRef } from '../../api';
+import { Progress } from '@backstage/core';
+import { Alert } from '@material-ui/lab';
+import TrendingDownIcon from '@material-ui/icons/TrendingDown';
+import TrendingUpIcon from '@material-ui/icons/TrendingUp';
+import TrendingFlatIcon from '@material-ui/icons/TrendingFlat';
+
+type Coverage = 'line' | 'branch';
+
+const getTrendIcon = (trend: number) => {
+  switch (true) {
+    case trend > 0:
+      return <TrendingUpIcon style={{ color: 'green' }} />;
+    case trend < 0:
+      return <TrendingDownIcon style={{ color: 'red' }} />;
+    case trend === 0:
+    default:
+      return <TrendingFlatIcon />;
+  }
+};
+
+export const CoverageHistoryChart = () => {
+  const { entity } = useEntity();
+  const codeCoverageApi = useApi(codeCoverageApiRef);
+  const {
+    loading: loadingHistory,
+    error: errorHistory,
+    value: valueHistory,
+  } = useAsync(
+    async () =>
+      await codeCoverageApi.getCoverageHistoryForEntity({
+        kind: entity.kind,
+        namespace: entity.metadata.namespace || 'default',
+        name: entity.metadata.name,
+      }),
+  );
+
+  if (loadingHistory) {
+    return <Progress />;
+  } else if (errorHistory) {
+    return <Alert severity="error">{errorHistory.message}</Alert>;
```

Comment:
[This one](https://github.com/backstage/backstage/blob/master/packages/core/src/components/ResponseErrorPanel/ResponseErrorPanel.tsx#L161) is a good error reporting panel to use, because it gives a ton of detail about errors when you want it. The catalog client [unpacks errors](https://github.com/backstage/backstage/blob/master/packages/catalog-client/src/CatalogClient.ts#L214) that are transferred by the [standard error handler middleware](https://github.com/backstage/backstage/blob/master/packages/backend-common/src/middleware/errorHandler.ts#L90). You could do the same unpacking in your client too and then use that panel